brigade.js

```diff
@@ -1,68 +1,152 @@
-  const docker = new Job("dind", "docker:stable-dind")
-  docker.privileged = true;
-  docker.env = {
-    DOCKER_DRIVER: driver
+class E2eJob extends Job {
+  constructor(name) {
+    super(name, "deis/node-chrome:node8");
+    this.tasks = [
+      "apt-get update -yq && apt-get install -yq --no-install-recommends udev ttf-freefont chromedriver chromium",
+      "cd /src",
+      "yarn global add @angular/cli",
+      "yarn install",
+      "ng e2e"
+    ];
   }
-  docker.tasks = [
-    "dockerd-entrypoint.sh &",
-    "sleep 20",
-    "cd /src",
-    "docker pull deis/kashti:canary || true",
-    "docker build -t deis/kashti:canary ."
-  ];
+}
 
-  // If a Docker user is specified, we push.
-  if (project.secrets.DOCKER_USER) {
-    docker.env.DOCKER_USER = project.secrets.DOCKER_USER
-    docker.env.DOCKER_PASS = project.secrets.DOCKER_PASS
-    docker.env.DOCKER_REGISTRY = project.secrets.DOCKER_REGISTRY
-    docker.tasks.push("docker login -u $DOCKER_USER -p $DOCKER_PASS $DOCKER_REGISTRY")
-    docker.tasks.push("docker push deis/kashti:angular5")
-  } else {
-    console.log("skipping push. DOCKER_USER is not set.");
+class ACRBuildJob extends Job {
+  constructor(name, img, tag, dir, registry, token, tenant) {
+    super(name, "microsoft/azure-cli:latest");
+    let imgName = img + ":" + tag;
+    this.env = {
+      AZURE_CONTAINER_REGISTRY: registry,
+      ACR_TOKEN: token,
+      ACR_TENANT: tenant,
+    }
+    this.tasks = [
+      // Create a service principal and assign it proper perms on the container registry.
+      `az login --service-principal -u $AZURE_CONTAINER_REGISTRY -p $ACR_TOKEN --tenant $ACR_TENANT`,
+      `cd ${dir}`,
+      `echo '========> building ${img}...'`,
+      `az acr build -r ${registry} -t ${imgName} .`,
+      `echo '<======== finished building ${img}.'`
+    ];
+  }
+}
+
+function ghNotify(state, msg, e, project) {
+  const gh = new Job(`notify-${state}`, "technosophos/github-notify:latest");
+  gh.env = {
+    GH_REPO: project.repo.name,
+    GH_STATE: state,
+    GH_DESCRIPTION: msg,
+    GH_CONTEXT: "brigade",
+    GH_TOKEN: project.secrets.ghToken,
+    GH_COMMIT: e.revision.commit
+  }
+  return gh
+}
+
+function test() {
+  const test = new TestJob(`${projectName}-test`)
+  const e2e = new E2eJob(`${projectName}-e2e`)
+  return Group.runAll([test, e2e]);
+}
+
+function checkRequested(e, p) {
+  console.log("check requested")
+  console.log(e.payload)
+  // Common configuration
+  const env = {
+    CHECK_PAYLOAD: e.payload,
+    CHECK_NAME: "Chart Tester",
+    CHECK_TITLE: "Lint all Helm Charts",
   }
 
-  // Run unit and e2e tests in parallel. Once both finish, run docker build.
-  Group.runAll([unitTests]).then(() => {
-    return docker.run()
-  });
-});
+  var tester = new TestJob(`${projectName}-test`)
+  var releaser = new ACRBuildJob(`${projectName}-test-release`, projectName, tag, "/src", project.secrets.acrName, project.secrets.acrToken, project.secrets.acrTenant);
```

Comment:
Where/how is `tag` set in this function?

Reply:
Good catch. `tag` is came from [line 122](https://github.com/Azure/kashti/pull/214/files#diff-a0d766ce006e9aa51ca74851ed4e6751R122) which is part of `githubRelease()`. In this particular section of code, `tag` is never set so it'll end up with a runtime error.

This section of code is all defunct and can probably be removed since [the brigade github app gateway](https://github.com/Azure/brigade-github-app) hasn't been installed on @technosophos's cluster. If you're interested in picking this up, lines 150-152 can be removed (removing the check suite events) and any related code there, then working from the `pull_request` event.